src/e2e/step-definitions/external-systems.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+
+import { Then, When } from "@cucumber/cucumber";
+import { Page } from "@playwright/test";
+import { CustomWorld } from "../support/worlds/world";
+
+const ONE_MINUTE_IN_MS = 60_000;
+
+// Reference to smartdocuments page
+let smartDocumentsPage: Page;
+
+When(
+  "Employee {string} clicks on Create Document for zaak",
+  { timeout: ONE_MINUTE_IN_MS },
+  async function (this: CustomWorld, user) {
+    await this.page.getByText("note_addDocument maken").click();
+
+    const sidebar = this.page.locator("div.sidenav-title");
+    await sidebar.waitFor({ state: "visible" });
+    await sidebar.getByText("Document maken");
+  },
+);
+
+When(
+  "Employee {string} fills in the create document form",
+  { timeout: ONE_MINUTE_IN_MS },
+  async function (this: CustomWorld, user) {
+    const submitButton = this.page.locator("#opslaan_button");
+    await submitButton.waitFor({ state: "visible" });
+
+    // Verify that the button is initially disabled
+    await submitButton.isDisabled();
+
+    // filling the create document form
+    await this.page.getByLabel("Sjabloongroep").click();
+    await this.page
+      .getByRole("option", { name: "Melding evenement organiseren behandelen" })
+      .click();
+    await this.page.waitForTimeout(1000);
+
+    await this.page.getByLabel("Sjabloon").last().click();
+    await this.page.getByRole("option", { name: "OpenZaakTest" }).click();
+
+    const autofillInputTitle = this.page.locator("#title_tekstfield");
+    await autofillInputTitle.click();
+    await autofillInputTitle.fill("Document Title Text");
+
+    // Now check if the button is enabled after those actions
+    await submitButton.isEnabled();
```

Comment:
What happens if the button is not, and will never be, enabled? Does it fail with a timeout?

Reply:
Yes I think so. Playwright has default timeouts for every await as far as I know.

Reply:
Playwright has a timeout to 'await'. Though your remark triggered me to find out. And then came to the conclusion that a previous cleanup had cleaned too much code. This is the correct code line:

`await this.expect(submitButton).toBeDisabled();`

There has to be an 'expect' to do the actual assertion.

Reply:
Fixed and works; Playwright wait 5 seconds it seems after testing it